data_input.py

```diff
@@ -2,47 +2,138 @@
 import os
 import pickle
+import glob
+
+import numpy as np
+import pandas as pd
 
 def tidy_most_recent(df, column='active'):
     df = df[column].reset_index().melt(id_vars='date')
     date_max = df['date'].max()
     df = df.query("date == @date_max")
-    return df
+    return df.sort_values('iso')
+
+
+
+
+MAP_UNMATCHED_COUNTRIES = {
+    'Bahamas, The': "Bahamas",
+    'The Bahamas': "Bahamas",
+    'Congo (Kinshasa)': "Democratic Republic of the Congo",
+    "Congo, the Democratic Republic of the":
+                            "Democratic Republic of the Congo",
+    'Congo (Brazzaville)': 'Republic of the Congo',
+    'Cape Verde': 'Cabo Verde',
+    "Czech Republic": 'Czechia',
+    "Cote d'Ivoire": 'Ivory Coast',
+    "Côte d'Ivoire": 'Ivory Coast',
+    "Swaziland": 'Eswatini',
+    'The Gambia': 'Gambia',
+    'Gambia, The': 'Gambia',
+    'Hong Kong SAR': "Hong Kong",
+    'Holy See': "Italy",
+    "Iran, Islamic Republic of": 'Iran',
+    'Iran (Islamic Republic of)': "Iran",
+    'Korea, South': "South Korea",
+    'Republic of Korea': "South Korea",
+    'Macau': 'Macao',
+    'Macao SAR': 'Macao',
+    'Mainland China': 'China',
+    "Moldova, Republic of": 'Moldova',
+    'Republic of Moldova': "Moldova",
+    'Republic of Ireland': 'Ireland',
+    "Macedonia, the former Yugoslav Republic of": 'North Macedonia',
+    "Réunion": 'Reunion',
+    "Russian Federation": "Russia",
+    "St. Martin": "Saint Martin",
+    'Taiwan*': "Taiwan",
+    'Taipei and environs': "Taiwan",
+    'Timor-Leste': 'Timor Leste',
+    'East Timor': 'Timor Leste',
+    'US': "United States",
+    'UK': 'United Kingdom',
+    'North Ireland': 'United Kingdom',
+    'occupied Palestinian territory': "Palestinian Territory",
+    'Palestine': "Palestinian Territory",
+    'Viet Nam': "Vietnam",
+    'Vatican City': 'Vatican',
+}
+
+UNMATCHED_COUNTRIES = ['Cruise Ship', 'Others', 'Channel Islands']
+
+
+def update_data():
+    """ Update the data source
+    """
+    os.chdir('COVID-19')
+    try:
+        os.system('git pull')
+    finally:
+        os.chdir('..')
 
 
 def get_data():
-    """ Download the data and return it as a 'wide' data frame
-    """
-    df = fetch_john_hopkins_data()
-    # The number of reported cases per day, country, and type
-    df_day = df.groupby(['country_region', 'iso', 'date', 'type']).sum()
-
-    # Switch to wide format (time series)
-    data = df_day.pivot_table(values='cases',
-                              columns=['type', 'iso', 'country_region'],
-                              index=['date'])
-    data = data.fillna(method='ffill')
-    data = data.fillna(value=0)
-
-    # Align columns and compute active cases
-    death, recovered = data['death'].align(
-        data['recovered'], join='outer', fill_value=0
-    )
-    inactive = death + recovered
-    confirmed, inactive = data['confirmed'].align(
-        inactive, join='outer', fill_value=0
-    )
-
-    active = confirmed - inactive
+    import data_input
```

Comment:
I don't understand this import here, since we are in the `data_input` module. Not blocking since from the artifact everything is ok.

Reply:
This is wrong idea. Thanks. Fixing.